camacq/gain.py

```diff
@@ -0,0 +1,228 @@
+                    else:
+                        green_sorted[green_val].append(key)
+                else:
+                    # Find the median value of all gains in
+                    # blue, yellow and red channels.
+                    mlist.append(int(val[i]))
+                    medians[channel] = int(np.median(mlist))
+        return {'green_sorted': green_sorted, 'medians': medians}
+
+    # #FIXME:10 Merge get_com and get_init_com functions, trello:egmsbuN8
+    def get_com(self, x_fields, y_fields, gain_result):
+        """Get command."""
+        dxcoord = 0
+        dycoord = 0
+        # Lists for storing command strings.
+        com_list = []
+        end_com_list = []
+        for gain, wells in gain_result['green_sorted'].iteritems():
+            end_com = []
+            channels = [gain,
+                        gain_result['medians']['blue'],
+                        gain_result['medians']['yellow'],
+                        gain_result['medians']['red']]
+            com = set_gain([], channels, self.job_list)
+            for well in sorted(wells):
+                for i in range(y_fields):
+                    for j in range(x_fields):
+                        # Only add selected fovs from file (arg) to cam list
+                        fov = '{}--X0{}--Y0{}'.format(well, j, i)
+                        if fov in self.coords.keys():
+                            dxcoord = self.coords[fov][0]
+                            dycoord = self.coords[fov][1]
+                            fov_is = True
+                        elif not self.coords:
+                            fov_is = True
+                        else:
+                            fov_is = False
+                        if fov_is:
+                            com.append(cam_com(self.pattern,
+                                               well,
+                                               'X0{}--Y0{}'.format(j, i),
+                                               dxcoord,
+                                               dycoord))
+                            end_com = ['CAM',
+                                       well,
+                                       'E0' + str(self.args.first_job + 2),
+                                       'X0{}--Y0{}'.format(j, i)]
+            # Store the commands in lists.
+            com_list.append(com)
+            end_com_list.append(end_com)
+        return {'com': com_list, 'end_com': end_com_list}
+
+    def get_init_com(self):
+        """Get command for gain analysis."""
+        wells = []
+        if self.template:
+            # Selected wells from template file.
+            wells = self.template.keys()
+        else:
+            # All wells.
+            for ucoord in range(int(get_wfx(self.args.last_well))):
+                for vcoord in range(int(get_wfy(self.args.last_well))):
+                    wells.append('U0' + str(ucoord) + '--V0' + str(vcoord))
```

Comment:
This will create a well called `U010--V00` etc for ucoord 10 and 11, ie for two digit integers. Use string formatting and pad a zero to single digit integers and no padding to double digit integers.

The current implementation will lead to error if no template is used.